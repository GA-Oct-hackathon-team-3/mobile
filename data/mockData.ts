export const mockFriend = {
  _id: "653b4a94d873b638dd4d0487",
  name: "Joe Dude",
  gender: "male",
  location: "NYC",
  dob: "2002-01-30",
  interests: [],
  tags: [
    {
      _id: "653f2ed1294e78fec9b4182d",
      type: "custom",
      title: "quirky",
      __v: 0,
    },
    {
      _id: "653f2ed1294e78fec9b41831",
      type: "custom",
      title: "minimal",
      __v: 0,
    },
  ],
  user: "65397add8bc93929acdd6df5",
  giftPreferences: ["present", "experience"],
  createdAt: "2023-10-27T05:28:52.530Z",
  updatedAt: "2023-10-30T17:54:28.864Z",
  __v: 2,
};
