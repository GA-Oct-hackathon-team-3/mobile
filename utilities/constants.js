export const WEB_BASE_URL =
  "https://presently-backend-64495929a7ac.herokuapp.com/api";
