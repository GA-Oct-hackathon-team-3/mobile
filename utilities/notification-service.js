import sendRequest from "./send-request";

const BASE_URL = "http://localhost:3010/api/device";

export async function acceptNotifications(tokenObj) {
  const res = await sendRequest(BASE_URL, "POST", tokenObj);

  console.log("res", res);
  return res;
}
