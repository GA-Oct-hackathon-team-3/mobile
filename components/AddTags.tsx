import React, { useRef, useState } from "react";
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
  TextInput,
  ScrollView,
  Button,
} from "react-native";
import { useRouter } from "expo-router";
import TitleBack from "./TitleBack";
import { colors } from "../constants/Theme";

export default function AddTags() {
  const [searchTag, setSearchTag] = useState("");
  const [addedTags, setAddedTags] = useState([
    "Pokemon",
    "Rock Climbing",
    "Chicago",
    "Male",
  ]);
  const tagCategories = {
    "Popular Tags": ["Movie Buff", "Minimal", "Quirky"],
    Aesthetic: ["Grunge", "Minimal", "Quirky"],
    Hobbies: [
      "Reading",
      "Outdoor Activities",
      "Arts and Crafts",
      "Socializing",
      "Sports",
      "Writing",
      "Working Out",
      "Cooking",
      "Gaming",
    ],
  };
  const router = useRouter();
  const scrollViewRef = useRef<ScrollView>();

  const handleTagPress = (tag) => {
    if (!addedTags.includes(tag)) {
      setAddedTags((prev) => [...prev, tag]);
    }
  };

  const handleSearchSubmit = () => {
    if (!addedTags.includes(searchTag) && searchTag !== "") {
      setAddedTags((prev) => [...prev, searchTag]);
      setSearchTag("");
    }
  };

  function scrollViewSizeChanged(height) {
    // y since we want to scroll vertically, use x and the width-value if you want to scroll horizontally
    scrollViewRef.current?.scrollTo({ y: height, animated: true });
  }

  return (
    <View style={styles.container}>
      <TitleBack title={"Add Tags"} />
      <View>
        <TextInput
          placeholder="Type to create custom tag"
          value={searchTag}
          onChangeText={setSearchTag}
          onSubmitEditing={handleSearchSubmit}
          style={styles.input}
        />

        <Text>Added Tags</Text>
        <ScrollView
          contentContainerStyle={{ maxHeight: 180 }}
          ref={scrollViewRef}
          onContentSizeChange={(width, height) => {
            scrollViewSizeChanged(height);
          }}
        >
          <View style={styles.addedTagsContainer}>
            {addedTags.map((tag) => (
              <Text key={tag} style={styles.tagSelectButton}>
                {tag}
              </Text>
            ))}
          </View>
        </ScrollView>

        {Object.entries(tagCategories).map(([category, tags]) => (
          <View key={category}>
            <Text>{category}</Text>
            <View style={styles.tagList}>
              {tags.map((tag) => (
                <TouchableOpacity
                  key={tag}
                  onPress={() => handleTagPress(tag)}
                  style={styles.tagButton}
                >
                  <Text>{tag} +</Text>
                </TouchableOpacity>
              ))}
            </View>
          </View>
        ))}
      </View>
      <View>
        <TouchableOpacity
          onPress={() => {
            router.push("/");
          }}
        >
          <View style={styles.button}>
            <Text style={styles.buttonText}>Complete Profile</Text>
          </View>
        </TouchableOpacity>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 20,
    flexDirection: "column",
    justifyContent: "space-between",
    backgroundColor: colors.cream,
  },
  header: {
    flexDirection: "row",
    alignItems: "center",
    marginBottom: 20,
  },
  headerText: {
    fontSize: 18,
    fontWeight: "bold",
  },
  input: {
    borderWidth: 1,
    borderColor: "#E0E0E0",
    padding: 10,
    marginBottom: 20,
  },
  addedTagsContainer: {
    flexDirection: "row",
    flexWrap: "wrap",
    marginBottom: 20,
  },
  tag: {
    borderWidth: 1,
    borderColor: "#E0E0E0",
    padding: 5,
    borderRadius: 20,
    margin: 5,
  },
  tagList: {
    flexDirection: "row",
    flexWrap: "wrap",
  },
  tagButton: {
    borderWidth: 1,
    padding: 15,
    borderRadius: 8,
    borderColor: "#D9D9D9",
    margin: 5,
  },
  tagSelectButton: {
    borderWidth: 1,
    padding: 15,
    borderRadius: 8,
    borderColor: "#D9D9D9",
    margin: 5,
    backgroundColor: "#D9D9D9",
  },
  button: {
    borderRadius: 20,
    backgroundColor: "#747474",
    alignItems: "center",
    justifyContent: "center",
    padding: 20,
  },
  buttonText: {
    color: "white",
  },
});
