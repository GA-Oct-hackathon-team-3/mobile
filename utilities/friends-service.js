import sendRequest from "./send-request";
import { getToken } from "./users-service";
import { WEB_BASE_URL } from "./constants";

const BASE_URL = `${WEB_BASE_URL}/friends`;

export async function retrieveFriends() {
  const friends = await sendRequest(BASE_URL, "GET", null);
  return friends;
}

export async function retrieveFriend(id) {
  const url = `${BASE_URL}/${id}`;

  const friend = await sendRequest(url, "GET", null);
  return friend;
}

export async function createFriend(friendData) {
  const newFriend = await sendRequest(`${BASE_URL}/create`, "POST", friendData);
  return newFriend;
}

export async function updateFriend(id, friendInput) {
  const response = await sendRequest(
    `${BASE_URL}/${id}/update`,
    "PUT",
    friendInput
  );
  return response;
}

export async function uploadPhoto(id, file) {
  console.log(id, typeof file, "ID AND FILE");

  const formData = new FormData();
  formData.append("photo", file);
  const token = getToken();

  const response = await fetch(`${BASE_URL}/${id}/upload`, {
    method: "POST",
    headers: {
      Authorization: `Bearer ${token}`,
      "Content-Type": "multipart/form-data",
    },
    body: formData,
  });

  console.log(response, "RESPONSE, UPLOAD PHOTO");

  if (response.status === 200) return response;
}
