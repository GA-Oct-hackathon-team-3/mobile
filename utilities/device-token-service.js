import { useState, useEffect, useRef } from 'react';
import * as Device from 'expo-device';
import * as Notifications from 'expo-notifications';
import Constants from 'expo-constants';
import { Platform } from 'react-native';
import * as notifService from './notification-service';


export const usePushNotifications = () => {
    Notifications.setNotificationHandler({
        handleNotification: async () => ({
          shouldShowAlert: true,
          shouldPlaySound: false,
          shouldSetBadge: false,
        }),
      });
}

export async function registerForPushNotificationsAsync ()  {
    let token;

        const { status : existingStatus } = await Notifications.getPermissionsAsync();
        let finalStatus = existingStatus;
        if (existingStatus !== 'granted') {
            const { status } = await Notifications.requestPermissionsAsync();
            finalStatus = status;
        }
        if (finalStatus !== 'granted') return;

        token = await Notifications.getExpoPushTokenAsync({
            projectId: Constants.expoConfig.extra.eas.projectId,
        });
        console.log(token);

    return token;
}