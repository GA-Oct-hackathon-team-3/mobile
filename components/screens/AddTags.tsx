import { useLocalSearchParams, useNavigation, useRouter } from "expo-router";
import React, { useEffect, useRef, useState } from "react";
import {
  ActivityIndicator,
  Image,
  ScrollView,
  StyleSheet,
  Text,
  TextInput,
  TouchableOpacity,
  View,
  useWindowDimensions,
} from "react-native";
import { colors } from "../../constants/Theme";
import * as friendsService from "../../utilities/friends-service";
import * as tagsService from "../../utilities/tags-service";
import TitleBack from "../TitleBack";
import { FontAwesome } from "@expo/vector-icons";
import TagAdder from "../TagAdder";

export default function AddTags() {
  const params = useLocalSearchParams();
  const navigation = useNavigation();

  const [searchTag, setSearchTag] = useState("");
  const [addedTags, setAddedTags] = useState([]);

  const [tags, setTags] = useState([]);
  const [defaultTags, setDefaultTags] = useState([]);
  const [suggestedTags, setSuggestedTags] = useState([]);
  const [loading, setLoading] = useState(false);
  const router = useRouter();
  const scrollViewRef = useRef<ScrollView>();
  const [friend, setFriend] = useState(null);
  const searchTagLower = searchTag.toLowerCase();
  const includesTag = addedTags.some(
    (tag) => tag.toLowerCase() === searchTagLower
  );

  const fetchTags = async () => {
    const friendData = await friendsService.retrieveFriend(params.id);
    setTags(friendData.tags)

    console.log(friendData);

    const tagsData = await tagsService.getTags();
    setDefaultTags(tagsData);
    console.log(tagsData);

    setAddedTags(friendData.tags);
    setFriend(friendData);
  };

  useEffect(() => {
    fetchTags();
  }, []);

  useEffect(() => {}, [addedTags]);

  const handleSearchSubmit = () => {
    if (!addedTags.includes(searchTag) && searchTag !== "") {
      setAddedTags((prev) => [...prev, searchTag]);
      setSearchTag("");
    }
  };

  function scrollViewSizeChanged(height) {
    // y since we want to scroll vertically, use x and the width-value if you want to scroll horizontally
    scrollViewRef.current?.scrollTo({ y: height, animated: true });
  }

  const handleSubmit = async () => {
    setLoading(true);

    try {
      addedTags.forEach(async (tag) => {
        await tagsService.addTag(friend._id, {
          title: tag,
          category: "Popular",
        });
      });
    } catch (err) {}
    setTimeout(() => {
      setLoading(false);
      navigation.dispatch({ type: "POP_TO_TOP" });
    }, 3000);
  };

  const handleTagPress = (tag) => {
    if (!addedTags.includes(tag)) {
      setAddedTags((prev) => [...prev, tag]);
    } else {
      setAddedTags((prev) => prev.filter((t) => t !== tag));
    }
  };

  return (
    <TagAdder tags={tags} setTags={setTags} defaultTags={defaultTags} />
  )
}