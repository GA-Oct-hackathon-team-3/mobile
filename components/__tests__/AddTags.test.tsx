import {
  render as rtlRender,
  cleanup,
  screen,
  act,
} from "@testing-library/react-native";
import AddTags from "../AddTags";
import { NavigationContainer } from "@react-navigation/native";
import fetch from "jest-fetch-mock";
import * as friendsService from "../../utilities/friends-service";
import { mockFriend } from "../../data/mockData";

jest.mock("../../utilities/friends-service");

describe("<AddTags />", () => {
  beforeEach(() => {
    fetch.resetMocks();
    // Mocking retrieveFriend
    friendsService.retrieveFriend.mockResolvedValue(mockFriend);
  });

  afterEach(() => {
    jest.clearAllMocks();
    cleanup();
  });

  it("renders correctly", async () => {
    let component;

    await act(async () => {
      component = rtlRender(<AddTags />);
    });

    expect(screen).toMatchSnapshot();
  });
});
