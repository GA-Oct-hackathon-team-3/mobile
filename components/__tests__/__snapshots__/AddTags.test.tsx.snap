// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddTags /> renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#FDF7ED",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "space-around",
    }
  }
>
  <View
    style={
      {
        "flexDirection": "column",
        "gap": 8,
        "maxHeight": 160,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingRight": 0,
          "width": 750,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Image
          source={1}
          style={
            {
              "height": 24,
              "left": -100,
              "width": 24,
            }
          }
        />
      </View>
      <Text
        style={
          {
            "fontFamily": "PilcrowMedium",
            "fontSize": 24,
          }
        }
      >
        Add Tags
      </Text>
    </View>
    <Text
      style={
        {
          "textAlign": "center",
        }
      }
    >
      Whatâ€™s your friend into? Adding tags helps Presently give more accurate gift suggestions.
    </Text>
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "center",
          "maxHeight": 200,
        }
      }
    >
      <Image
        source={1}
        style={
          {
            "height": 100,
            "width": 100,
          }
        }
      />
      <Image
        source={1}
        style={
          {
            "height": 130,
            "transform": [
              {
                "rotateY": "180deg",
              },
            ],
            "width": 100,
          }
        }
      />
      <Image
        source={1}
        style={
          {
            "height": 100,
            "transform": [
              {
                "rotateY": "180deg",
              },
            ],
            "width": 100,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "padding": 20,
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Type to create custom tag"
      style={
        {
          "backgroundColor": "#FFF",
          "borderColor": "#E0E0E0",
          "borderRadius": 10,
          "borderWidth": 1,
          "marginBottom": 20,
          "padding": 10,
        }
      }
      value=""
    />
    <Text>
      Added Tags
    </Text>
    <RCTScrollView
      contentContainerStyle={
        {
          "maxHeight": 180,
        }
      }
      onContentSizeChange={[Function]}
    >
      <View>
        <View
          style={
            {
              "flexDirection": "row",
              "flexWrap": "wrap",
              "marginBottom": 20,
            }
          }
        >
          <Text
            style={
              {
                "backgroundColor": "#AF95E7",
                "borderColor": "#D9D9D9",
                "borderRadius": 8,
                "borderWidth": 1,
                "color": "#FFF",
                "fontFamily": "PilcrowRounded",
                "margin": 5,
                "opacity": 0.95,
                "padding": 15,
              }
            }
          >
            Pokemon
          </Text>
          <Text
            style={
              {
                "backgroundColor": "#AF95E7",
                "borderColor": "#D9D9D9",
                "borderRadius": 8,
                "borderWidth": 1,
                "color": "#FFF",
                "fontFamily": "PilcrowRounded",
                "margin": 5,
                "opacity": 0.95,
                "padding": 15,
              }
            }
          >
            Rock Climbing
          </Text>
          <Text
            style={
              {
                "backgroundColor": "#AF95E7",
                "borderColor": "#D9D9D9",
                "borderRadius": 8,
                "borderWidth": 1,
                "color": "#FFF",
                "fontFamily": "PilcrowRounded",
                "margin": 5,
                "opacity": 0.95,
                "padding": 15,
              }
            }
          >
            Chicago
          </Text>
          <Text
            style={
              {
                "backgroundColor": "#AF95E7",
                "borderColor": "#D9D9D9",
                "borderRadius": 8,
                "borderWidth": 1,
                "color": "#FFF",
                "fontFamily": "PilcrowRounded",
                "margin": 5,
                "opacity": 0.95,
                "padding": 15,
              }
            }
          >
            Male
          </Text>
        </View>
      </View>
    </RCTScrollView>
    <RCTScrollView
      contentContainerStyle={
        {
          "maxHeight": 240,
          "paddingBottom": 100,
        }
      }
    >
      <View>
        <View>
          <Text>
            Popular Tags
          </Text>
          <View
            style={
              {
                "flexDirection": "row",
                "flexWrap": "wrap",
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#FFF",
                  "borderColor": "#D9D9D9",
                  "borderRadius": 8,
                  "borderWidth": 1,
                  "margin": 5,
                  "opacity": 1,
                  "padding": 15,
                }
              }
            >
              <Text
                style={
                  {
                    "fontFamily": "PilcrowRounded",
                  }
                }
              >
                Movie Buff
                 +
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#FFF",
                  "borderColor": "#D9D9D9",
                  "borderRadius": 8,
                  "borderWidth": 1,
                  "margin": 5,
                  "opacity": 1,
                  "padding": 15,
                }
              }
            >
              <Text
                style={
                  {
                    "fontFamily": "PilcrowRounded",
                  }
                }
              >
                Minimal
                 +
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#FFF",
                  "borderColor": "#D9D9D9",
                  "borderRadius": 8,
                  "borderWidth": 1,
                  "margin": 5,
                  "opacity": 1,
                  "padding": 15,
                }
              }
            >
              <Text
                style={
                  {
                    "fontFamily": "PilcrowRounded",
                  }
                }
              >
                Quirky
                 +
              </Text>
            </View>
          </View>
        </View>
        <View>
          <Text>
            Aesthetic
          </Text>
          <View
            style={
              {
                "flexDirection": "row",
                "flexWrap": "wrap",
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#FFF",
                  "borderColor": "#D9D9D9",
                  "borderRadius": 8,
                  "borderWidth": 1,
                  "margin": 5,
                  "opacity": 1,
                  "padding": 15,
                }
              }
            >
              <Text
                style={
                  {
                    "fontFamily": "PilcrowRounded",
                  }
                }
              >
                Grunge
                 +
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#FFF",
                  "borderColor": "#D9D9D9",
                  "borderRadius": 8,
                  "borderWidth": 1,
                  "margin": 5,
                  "opacity": 1,
                  "padding": 15,
                }
              }
            >
              <Text
                style={
                  {
                    "fontFamily": "PilcrowRounded",
                  }
                }
              >
                Minimal
                 +
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#FFF",
                  "borderColor": "#D9D9D9",
                  "borderRadius": 8,
                  "borderWidth": 1,
                  "margin": 5,
                  "opacity": 1,
                  "padding": 15,
                }
              }
            >
              <Text
                style={
                  {
                    "fontFamily": "PilcrowRounded",
                  }
                }
              >
                Quirky
                 +
              </Text>
            </View>
          </View>
        </View>
        <View>
          <Text>
            Hobbies
          </Text>
          <View
            style={
              {
                "flexDirection": "row",
                "flexWrap": "wrap",
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#FFF",
                  "borderColor": "#D9D9D9",
                  "borderRadius": 8,
                  "borderWidth": 1,
                  "margin": 5,
                  "opacity": 1,
                  "padding": 15,
                }
              }
            >
              <Text
                style={
                  {
                    "fontFamily": "PilcrowRounded",
                  }
                }
              >
                Reading
                 +
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#FFF",
                  "borderColor": "#D9D9D9",
                  "borderRadius": 8,
                  "borderWidth": 1,
                  "margin": 5,
                  "opacity": 1,
                  "padding": 15,
                }
              }
            >
              <Text
                style={
                  {
                    "fontFamily": "PilcrowRounded",
                  }
                }
              >
                Outdoor Activities
                 +
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#FFF",
                  "borderColor": "#D9D9D9",
                  "borderRadius": 8,
                  "borderWidth": 1,
                  "margin": 5,
                  "opacity": 1,
                  "padding": 15,
                }
              }
            >
              <Text
                style={
                  {
                    "fontFamily": "PilcrowRounded",
                  }
                }
              >
                Arts and Crafts
                 +
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#FFF",
                  "borderColor": "#D9D9D9",
                  "borderRadius": 8,
                  "borderWidth": 1,
                  "margin": 5,
                  "opacity": 1,
                  "padding": 15,
                }
              }
            >
              <Text
                style={
                  {
                    "fontFamily": "PilcrowRounded",
                  }
                }
              >
                Socializing
                 +
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#FFF",
                  "borderColor": "#D9D9D9",
                  "borderRadius": 8,
                  "borderWidth": 1,
                  "margin": 5,
                  "opacity": 1,
                  "padding": 15,
                }
              }
            >
              <Text
                style={
                  {
                    "fontFamily": "PilcrowRounded",
                  }
                }
              >
                Sports
                 +
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#FFF",
                  "borderColor": "#D9D9D9",
                  "borderRadius": 8,
                  "borderWidth": 1,
                  "margin": 5,
                  "opacity": 1,
                  "padding": 15,
                }
              }
            >
              <Text
                style={
                  {
                    "fontFamily": "PilcrowRounded",
                  }
                }
              >
                Writing
                 +
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#FFF",
                  "borderColor": "#D9D9D9",
                  "borderRadius": 8,
                  "borderWidth": 1,
                  "margin": 5,
                  "opacity": 1,
                  "padding": 15,
                }
              }
            >
              <Text
                style={
                  {
                    "fontFamily": "PilcrowRounded",
                  }
                }
              >
                Working Out
                 +
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#FFF",
                  "borderColor": "#D9D9D9",
                  "borderRadius": 8,
                  "borderWidth": 1,
                  "margin": 5,
                  "opacity": 1,
                  "padding": 15,
                }
              }
            >
              <Text
                style={
                  {
                    "fontFamily": "PilcrowRounded",
                  }
                }
              >
                Cooking
                 +
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#FFF",
                  "borderColor": "#D9D9D9",
                  "borderRadius": 8,
                  "borderWidth": 1,
                  "margin": 5,
                  "opacity": 1,
                  "padding": 15,
                }
              }
            >
              <Text
                style={
                  {
                    "fontFamily": "PilcrowRounded",
                  }
                }
              >
                Gaming
                 +
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#53CF85",
            "borderRadius": 20,
            "height": 60,
            "justifyContent": "center",
            "width": "60%",
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontFamily": "PilcrowRounded",
              "fontSize": 20,
            }
          }
        >
          Complete Profile
        </Text>
      </View>
    </View>
  </View>
</View>
`;
